pyvrp/cpp/search/LocalSearch.cpp

```diff
@@ -383,9 +397,24 @@ void LocalSearch::insert(Route::Node *U,
     {
         UAfter->route()->insert(UAfter->idx() + 1, U);
         update(UAfter->route(), UAfter->route());
+        markPromising(U);
     }
 }
 
+void LocalSearch::markPromising(Route::Node *U)
+{
+    assert(U->route());
+
+    if (!U->isDepot())
+        promising[U->client()] = true;
+
+    if (!U->isStartDepot() && !p(U)->isDepot())
+        promising[p(U)->client()] = true;
+
+    if (!U->isEndDepot() && !n(U)->isDepot())
+        promising[n(U)->client()] = true;
```

Comment:
Does it matter if `U`, `p(U)`, or `n(U)` are depots? We never evaluate those, right?